componentLibraries/defaultLibrary/Signal/Sources&Sinks/SignalLogger.hpp

```diff
@@ -0,0 +1,258 @@
+            //(only important for Hopsan row-wise CSV, where aliases and units are used)
+            if(mNumVars != namesVector.size()) {
+                stopSimulation("Number of variable names must equal number of input variables");
+                return;
+            }
+            if(mFileType == HopsanRowWiseCSV && mNumVars != aliasesVector.size()) {
+                stopSimulation("Number of variable aliases must equal number of input variables");
+                return;
+            }
+            if(mFileType == HopsanRowWiseCSV && mNumVars != unitsVector.size()) {
+                stopSimulation("Number of variable units must equal number of input variables");
+                return;
+            }
+
+            //Open file for writing
+            mFile.open(mFilePath.c_str());
+            if (!mFile.is_open()) {
+                stopSimulation("Could not open file for writing: "+ mFilePath);
+                return;
+            }
+
+            //Write header lines/rows depending on file type
+            if(mFileType == PlainColumnWiseCSV) {
+                //First row should be variable names
+                mFile << "Time,";
+                for(size_t i=0; i<mNumVars; ++i) {
+                    mFile << namesVector[i].c_str();
+                    if(i < mNumVars-1) {
+                        mFile << ",";
+                    }
+                }
+                mFile << "\n";
+            }
+            else if(mFileType == PlainRowWiseCSV) {
+                //First column should be variable names
+                //Create one row for each variable and insert names first
+                mRows.resize(mNumVars+1);
+                mRows[0].append("Time");
+                for(size_t i=0; i<mNumVars; ++i) {
+                    mRows[i+1].append(namesVector[i]);
+                }
+            }
+            else if(mFileType == HopsanRowWiseCSV) {
+                //First three columns should be variable names, aliases and units
+                //Create one row for each variable and insert this first
+                mRows.resize(mNumVars+1);
+                mRows[0].append("Time,,");
+                for(size_t i=0; i<mNumVars; ++i) {
+                    mRows[i+1].append(namesVector[i]+",");
+                    mRows[i+1].append(aliasesVector[i]+",");
+                    mRows[i+1].append(unitsVector[i]);
+                }
+            }
+            if(mFileType == HopsanPLO) {
+                //Write custom PLO header rows
+
+                //Extract target file name
+                HVector<HString> temp = mFilePath.split('/');
+                temp = temp[temp.size()-1].split('\\');
+                HString fileName = temp[temp.size()-1];
+
+                //Fetch parent system name (hmf filename is unknown)
+                HString modelName = getSystemParent()->getName();
+
+                //Generate date and time string
+                auto t = std::time(nullptr);
```

Comment:
Lots of c++11 in this file, then we need to drop support for old MSVC compilers, and maybe really old GCC as well.
Would require dropping SIT export support also. But maybe not such bad thing ?

Reply:
I removed all C++11 stuff I think (auto, to_string and date/time stuff). I hope I did not miss anything. People are still using SIT export, so we should probably wait with allowing C++11 in (default) components.

Reply:
OK